### PR TITLE
update doc for Oracle translate function

### PR DESCRIPTION
**Pull request by wzru**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

To add doc for new feature, supporting Oracle translate function

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/6053
- Other reference link(s): https://github.com/pingcap/tidb/pull/26408

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by XuHuaiyu — APPROVED** [review]
LGTM

File: functions-and-operators/string-functions.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ aliases: ['/docs-cn/dev/functions-and-operators/string-functions/','/docs-cn/dev
 
 # 字符串函数
 
-TiDB 支持使用 MySQL 5.7 中提供的大部分[字符串函数](https://dev.mysql.com/doc/refman/5.7/en/string-functions.html)。
+TiDB 支持使用 MySQL 5.7 中提供的大部分[字符串函数](https://dev.mysql.com/doc/refman/5.7/en/string-functions.html)和 Oracle 21 中提供的一些[函数](https://docs.oracle.com/en/database/oracle/oracle-database/21/sqlqr/SQL-Functions.html#GUID-93EC62F8-415D-4A7E-B050-5D5B2C127009)。
```
Comment (TomShawn) [inline functions-and-operators/string-functions.md:8]: 是说 Oracle 这个文档网页上列的所有函数 TiDB 都支持吗？

File: functions-and-operators/string-functions.md [hunk 2]
```diff
@@ -59,6 +59,7 @@ TiDB 支持使用 MySQL 5.7 中提供的大部分[字符串函数](https://dev.m
 | [`SUBSTRING()`](https://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_substring)  | 返回指定的子字符串   |
 | [`SUBSTRING_INDEX()`](https://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_substring-index)  | 从一个字符串中返回指定出现次数的定界符之前的子字符串  |
 | [`TO_BASE64()`](https://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_to-base64)  | 返回转化为 base-64 表示的字符串参数   |
+| [`TRANSLATE()`](https://docs.oracle.com/en/database/oracle/oracle-database/21/sqlrf/TRANSLATE.html#GUID-80F85ACB-092C-4CC7-91F6-B3A585E3A690) | 将字符串中出现的所有指定的字符替换为其它的字符。这个函数不会像 Oracle 一样将空字符串视为`NULL` |
```
Comment (TomShawn) [inline functions-and-operators/string-functions.md:62]:
```suggestion
| [`TRANSLATE()`](https://docs.oracle.com/en/database/oracle/oracle-database/21/sqlrf/TRANSLATE.html#GUID-80F85ACB-092C-4CC7-91F6-B3A585E3A690) | 将字符串中出现的所有指定字符替换为其它字符。这个函数不会像 Oracle 一样将空字符串视为`NULL` |
```

File: functions-and-operators/string-functions.md (hunk 1 above)
Comment (wzru) [reply functions-and-operators/string-functions.md:8]: 只支持所列的部分函数

Comment (TomShawn) [reply functions-and-operators/string-functions.md:8]:
```suggestion
TiDB 支持使用大部分 MySQL 5.7 中提供的[字符串函数](https://dev.mysql.com/doc/refman/5.7/en/string-functions.html)和一部分 Oracle 21 所提供的[函数](https://docs.oracle.com/en/database/oracle/oracle-database/21/sqlqr/SQL-Functions.html#GUID-93EC62F8-415D-4A7E-B050-5D5B2C127009)。
```

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**